zeppelin-user-impersonation[3016].md

```diff
@@ -0,0 +1,152 @@
+###Sample LDAP configuration, for user Authentication, currently tested for single Realm </br>
+[main] </br>
+###A sample for configuring Active Directory Realm </br>
+activeDirectoryRealm = org.apache.zeppelin.realm.ActiveDirectoryGroupRealm </br>
+activeDirectoryRealm.systemUsername = CN=Administrator,CN=Users,DC=ADSERVER,DC=com </br>
+activeDirectoryRealm.systemPassword = loginpassword </br>
+activeDirectoryRealm.searchBase = CN=Users,DC=ADSERVER,DC=com </br>
+activeDirectoryRealm.url = ldap://ldap.ad.ADSERVER.com:389 </br>
+activeDirectoryRealm.authorizationCachingEnabled = true </br>
+sessionManager = org.apache.shiro.web.session.mgt.DefaultWebSessionManager </br>
+securityManager.sessionManager = $sessionManager </br>
+
+###If caching of user is required then uncomment below lines </br>
+cacheManager = org.apache.shiro.cache.MemoryConstrainedCacheManager </br>
+securityManager.cacheManager = $cacheManager </br>
+###securityManager.sessionManager = $sessionManager </br>
+###86,400,000 milliseconds = 24 hour </br>
+securityManager.sessionManager.globalSessionTimeout = 86400000 </br>
+shiro.loginUrl = /api/login </br>
+[roles] </br>
+role1 = * </br>
+role2 = * </br>
+role3 = * </br>
+admin = * </br>
+[urls] </br>
+###This section is used for url-based security. </br>
+###You can secure interpreter, configuration and credential information by urls. Comment or uncomment the below urls that you want to hide. </br>
+###anon means the access is anonymous. </br>
+###authc means Form based Auth Security </br>
+###To enfore security, comment the line below and uncomment the next one </br>
+/api/version = anon </br>
+#/api/interpreter/** = authc, roles[admin] </br>
+#/api/configurations/** = authc, roles[admin] </br>
+#/api/credential/** = authc, roles[admin] </br>
+#/** = anon </br>
+/** = authc </br>
+</i>
+
+<b>Below are the configurations required from the environment:</b> </br>
+•	On the Zeppelin server node make sure to add the below to <i>/etc/sudoers</i>: </br>
+With the root user: </br>
+
+`$visudo` </br>
+`zeppelin        ALL=(ALL)       NOPASSWD: ALL ` </br> 
+
+If this entry is missing in the /etc/sudoers then the interpreter fails with the below error: </br>
+
+`org.apache.zeppelin.interpreter.InterpreterException: Permission denied, please try again.`</br>
+`Permission denied, please try again.` </br>
+`Permission denied (publickey,gssapi-keyex,gssapi-with-mic,password).`</br> 
+</br>
+
+•	Log folder (usually /var/log/zeppelin or /var/log/hadoop/zeppelin) of zeppelin should be of 777 instead of 755 if not the Spark interpreter keep failing with below error as the impersonated user won’t able to write to the logs folder:
+
+`java.lang.NullPointerException`</br>
+	`at org.apache.zeppelin.spark.Utils.invokeMethod(Utils.java:38)`</br>
+	`at org.apache.zeppelin.spark.Utils.invokeMethod(Utils.java:33)` </br>
+
+</br>
+•	If you are on a HDP platform this can’t be achieved from the CLI which needs a Ambari Zeppelin code level change </br>
```

Comment:
I don't think we need to cover this here, changing ambari source code is not a feasible solution for general users. I would recommend to write hcc article about this in hortonworks community forum.

Reply:
Sure, will address this.